L1Trigger/L1TMuonOverlapPhase2/interface/InputMakerPhase2.h

```diff
@@ -0,0 +1,116 @@
+#define INTERFACE_INPUTMAKERPHASE2_H_
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/L1DTTrackFinder/interface/L1MuDTChambThContainer.h"
+#include "DataFormats/L1DTTrackFinder/interface/L1Phase2MuDTPhContainer.h"
+#include "DataFormats/L1TMuon/interface/RegionalMuonCandFwd.h"
+#include "DataFormats/MuonDetId/interface/DTChamberId.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+
+#include "L1Trigger/L1TMuonOverlapPhase1/interface/Omtf/OMTFConfiguration.h"
+#include "L1Trigger/L1TMuonOverlapPhase1/interface/MuonStub.h"
+#include "L1Trigger/L1TMuonOverlapPhase1/interface/MuonStubMakerBase.h"
+#include "L1Trigger/L1TMuonOverlapPhase1/interface/Omtf/OMTFinputMaker.h"
+#include "L1Trigger/L1TMuonOverlapPhase2/interface/OmtfPhase2AngleConverter.h"
+
+class DtPhase2DigiToStubsConverter : public DigiToStubsConverterBase {
+public:
+  DtPhase2DigiToStubsConverter(edm::EDGetTokenT<L1Phase2MuDTPhContainer> inputTokenDtPh,
+                               edm::EDGetTokenT<L1MuDTChambThContainer> inputTokenDtTh)
+      : inputTokenDtPh(inputTokenDtPh), inputTokenDtTh(inputTokenDtTh){};
+
+  ~DtPhase2DigiToStubsConverter() override{};
+
+  //virtual void initialize(const edm::ParameterSet& edmCfg, const edm::EventSetup& es, const ProcConfigurationBase* procConf) {};
+
+  void loadDigis(const edm::Event& event) override;
+
+  void makeStubs(MuonStubPtrs2D& muonStubsInLayers,
+                 unsigned int iProcessor,
+                 l1t::tftype procTyp,
+                 int bxFrom,
+                 int bxTo,
+                 std::vector<std::unique_ptr<IOMTFEmulationObserver> >& observers) override;
+
+  //dtThDigis is provided as argument, because in the OMTF implementation the phi and eta digis are merged (even thought it is artificial)
+  virtual void addDTphiDigi(MuonStubPtrs2D& muonStubsInLayers,
+                            const L1Phase2MuDTPhDigi& digi,
+                            const L1MuDTChambThContainer* dtThDigis,
+                            unsigned int iProcessor,
+                            l1t::tftype procTyp) = 0;
+
+  virtual void addDTetaStubs(MuonStubPtrs2D& muonStubsInLayers,
+                             const L1MuDTChambThDigi& thetaDigi,
+                             unsigned int iProcessor,
+                             l1t::tftype procTyp) = 0;
+
+  virtual bool acceptDigi(const DTChamberId& dTChamberId, unsigned int iProcessor, l1t::tftype procType) {
+    return true;
+  }
+
+protected:
+  bool mergePhiAndTheta = true;
+
+  edm::EDGetTokenT<L1Phase2MuDTPhContainer> inputTokenDtPh;
+  edm::EDGetTokenT<L1MuDTChambThContainer> inputTokenDtTh;
+
+  edm::Handle<L1Phase2MuDTPhContainer> dtPhDigis;
+  edm::Handle<L1MuDTChambThContainer> dtThDigis;
+};
+
+class DtPhase2DigiToStubsConverterOmtf : public DtPhase2DigiToStubsConverter {
+public:
+  DtPhase2DigiToStubsConverterOmtf(const OMTFConfiguration* config,
+                                   const OmtfAngleConverter* angleConverter,
```

Comment:
This doesn't seem possible due to the amount of underlying code that needs to use these (so don't worry about this), but I don't suppose either of these could be references, and not have to worry about having `nullptr`s around?